### PR TITLE
Make sure the Docker Host IP address is retrieved correctly when running inside docker

### PR DESCRIPTION
**Pull request by jochenhebbrecht**
More details can be found here: https://github.com/testcontainers/testcontainers-java/issues/479#issuecomment-356895127

### CODE REVIEW

File: core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java [hunk 1]
```diff
@@ -23,10 +25,15 @@
                     cmd -> cmd.withCmd("sh", "-c", "ip route|awk '/default/ { print $3 }'"),
                     (client, id) -> {
                         try {
-                            return client.logContainerCmd(id)
-                                    .withStdOut(true)
-                                    .exec(new LogToStringContainerCallback())
-                                    .toString();
+                            LogToStringContainerCallback loggingCallback = new LogToStringContainerCallback();
+                            client.logContainerCmd(id).withStdOut(true)
+                                                      .withStdErr(true)
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java]: I think `withStdErr` is not needed here

File: core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java [hunk 2]
```diff
@@ -23,10 +25,15 @@
...
+                                                      .withFollowStream(true)
+                                                      .withTailAll()
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java]: `tail=all` is the default, can also remove it

File: core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java [hunk 3]
```diff
@@ -23,10 +25,15 @@
                     cmd -> cmd.withCmd("sh", "-c", "ip route|awk '/default/ { print $3 }'"),
                     (client, id) -> {
                         try {
-                            return client.logContainerCmd(id)
-                                    .withStdOut(true)
-                                    .exec(new LogToStringContainerCallback())
-                                    .toString();
+                            LogToStringContainerCallback loggingCallback = new LogToStringContainerCallback();
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/dockerclient/DockerClientConfigUtils.java]: please mention your fix in `CHANGELOG.md` 👍

**Comment by kiview** [comment]
@bsideup I'm afraid this fix won't help with the flaky log behavior regarding `LogMessageWaitStrategy`, since we use a different class there (`LogUtils`) and are already using `withFollowStream(true)`:
https://github.com/testcontainers/testcontainers-java/blob/c2eae85232653273748aaa4f5f765f48c28e0f87/core/src/main/java/org/testcontainers/utility/LogUtils.java#L26

Or did you have another idea?

**Comment by bsideup** [comment]
@kiview this PR is not about `LogMessageWaitStrategy`... like... at all :)

LogMessageWaitStrategy's issue seems to be related to some race conditions

**Comment by kiview** [comment]
Yes I know, I was referring to the discussion in https://github.com/testcontainers/testcontainers-java/issues/479#issuecomment-356889324, maybe I mixed this up :wink:

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,7 @@ All notable changes to this project will be documented in this file.
 ### Fixed
 - Fixed problem with case-sensitivity when checking internal port. ([\#524](https://github.com/testcontainers/testcontainers-java/pull/524))
 - Add retry logic around checkExposedPort pre-flight check for improved robustness ([\#513](https://github.com/testcontainers/testcontainers-java/issues/513))
+- Fixed retrieval of Docker host IP when running inside Docker. ([\#479](https://github.com/testcontainers/testcontainers-java/issues/479))
```
Comment (bsideup) [inline CHANGELOG.md]: This should go to the "upcoming" section, I will create one :)

Comment (jochenhebbrecht) [reply CHANGELOG.md]: Urgh ... sorry, my mistake. Sorry, was doing things too fast. Thanks for taking this one on you.

**Comment by bsideup** [comment]
@jochenhebbrecht merged, thanks! 👍 💪

**Pull request merged**